sawmill-core/src/main/java/io/logz/sawmill/Pipeline.java

```diff
@@ -38,11 +38,13 @@ public boolean isStopOnFailure() {
 
         private final ExecutionStepsParser executionStepsParser;
 
-        public Factory() {
-            this(ProcessorFactoryRegistry.getInstance(), ConditionFactoryRegistry.getInstance());
+        public Factory(PipelineConfiguration...pipelineConfigurations) {
+            this(new ProcessorFactoryRegistry(new ProcessorFactoriesLoader(new TemplateService(), pipelineConfigurations)),
+                    new ConditionFactoryRegistry(new ConditionalFactoriesLoader(new TemplateService(), pipelineConfigurations)));
         }
```

Comment:
in this case we will violate the DI principle and initialize the dependencies inline which is bad to me in terms of unit testing and coupling

Reply:
I think it will make the API more simple - there is no reason for a user to pass anything but `new TemplateService()` to the `ProcessorFactoriesLoader` constructor. Thus, I see no harm by inlining the instantiation of `TemplateService`

Reply:
Sorry, Naor, but I disagree

to me that violates the DI and makes code less maintainable & more coupled

also unit tests in that case are harder to implement since you need to re-set the dependencies which are inlined